django/views/defaults.py

```diff
@@ -59,12 +74,19 @@ def server_error(request, template_name='500.html'):
     try:
         template = loader.get_template(template_name)
     except TemplateDoesNotExist:
+        if template_name != ERROR_500_TEMPLATE_NAME:
+            warnings.warn(
+                "A template_name has been provided, but the template cannot be found. "
```

Comment:
@carljm @timgraham to me it's perfectly fine to skip a deprecation path and just return an error if the wrong argument is given
If there is consensus on this, I will happily change the PR